### PR TITLE
Adding in User Creation for outbound queue

### PR DESCRIPTION
**Pull request by nadiabahrami**
Signed-off-by: Nadia Bahrami <nadia.bahrami@dev9.com>

### CODE REVIEW

File: rbac/providers/azure/delta_outbound_sync.py [hunk 1]
```diff
@@ -162,22 +165,35 @@ def create_entry_aad(queue_entry):
...
+        else:
+            LOGGER.warning("Unable to create user in AAD: %s", queue_entry)
+            raise ExpectedError("Unable to create user.")
 
 
 def create_group_aad(queue_entry):
     """Creates a given group in aad."""
     # TODO: Implement group creation in Azure AD. Currently logs and deletes
     #   entry and throws an ExpectedError.
+    LOGGER.info("You are in create_group_aad")
```
Comment (mtn217) [inline rbac/providers/azure/delta_outbound_sync.py]: Would this log be better in debug? As we are trying to not spam our logging output.

File: rbac/providers/common/outbound_filters.py [hunk 2]
```diff
@@ -43,6 +45,29 @@ def outbound_group_filter(sawtooth_group, provider):
...
+
+
+def outbound_user_creation_filter(sawtooth_user, provider):
+    """Takes in a user dict from queue_outbound and formats it to a provider's specs.
+    :param: user > dict > a dictionary representing a user
+    :param: provider > str > inbound provider type (azure, ldap)
+    """
+    if provider != "azure" and provider != "ldap":
+        raise TypeError("Provider must be specified with a valid option.")
+    aad_user = {}
```
Comment (mtn217) [inline rbac/providers/common/outbound_filters.py]: Seems we are using both providers, perhaps just change this to just `user = {}`

File: rbac/providers/azure/delta_outbound_sync.py (hunk 1 above)
Comment (nadiabahrami) [reply rbac/providers/azure/delta_outbound_sync.py]: Sorry I am using that for the next PR for testing.  I will remove.

File: rbac/providers/common/outbound_filters.py (hunk 2 above)
Comment (nadiabahrami) [reply rbac/providers/common/outbound_filters.py]: Ah yes will change

**Pull request merged**